apps/web/vibes/soul/primitives/cursor-pagination/index.tsx

```diff
@@ -15,7 +17,10 @@ export interface CursorPaginationInfo {
 }
 
 interface Props {
-  info: CursorPaginationInfo | Promise<CursorPaginationInfo>;
+  label?: Streamable<string | null>;
+  info: Streamable<CursorPaginationInfo>;
+  previousLabel?: Streamable<string | null>;
+  nextLabel?: Streamable<string | null>;
```

Comment:
Do you have an example of why these label values are streamable?

Reply:
It's because these are translated and getting translations is asynchronous

Reply:
As I was implementing changes to the category pages in Catalyst I realized that labels need to be streamable for this reason

Reply:
So do we need to make all labels Streamable?

Reply:
Yeah, probably. I'm planning on doing this as-needed as I work on Catalyst and VIBES